builder/proxmox/common/config_test.go

```diff
@@ -244,7 +245,47 @@ func TestAdditionalISOs(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestDeprecatedISOOptionsAreConverted(t *testing.T) {
+	isotests := []struct {
+		name string
+		ISOs map[string]interface{}
+	}{
+		{
+			// Ensure deprecated device field is converted
+			name: "device should be converted to type and index",
+			ISOs: map[string]interface{}{
+				"device":   "ide3",
```

Comment:
Might be superfluous, but I wonder if we should add cases with a device index too high? These are probably caught by the blob above though, I'll leave it up to you to decide

Reply:
The handling of bus overallocation moved to `generateProxmoxDisks` in `builder/proxmox/common/step_start_vm.go` because there's also the potential for overallocation when preserving clone builder disks.

I've pushed an update for `TestGenerateProxmoxDisks` in `builder/proxmox/common/step_start_vm_test.go` to test overallocating sata storage with a mix of defined disks, ISOs and clone source vms mappings, do you think one is needed for each storage type?